mlir/lib/Conversion/AIRToAIEPass.cpp

```diff
@@ -151,24 +288,18 @@ struct DMAAllocator {
     }
   }
 
-  AIE::TileOp getTile(AIE::DeviceOp aie_device, air::DmaMemcpyInterface &dmaOp,
+  template <typename T>
+  AIE::TileOp getTile(AIE::DeviceOp aie_device, T &memcpyOp,
```

Comment:
It has occurred to me in the past that we could unify some or all of the ChannelInterface and DmaMemcpyInterface which is basically what's going on here via the template and all the new helper functions like `getSizesInterface` and `isTileInbound/isTileOutbound`.

Reply:
I agree with you--unifying the two interfaces makes sense as they are almost the same. I will unify them in a future PR so that this PR only focuses on getting channel-to-locks pass working.